src/main/java/org/testng/internal/ClassImpl.java

```diff
@@ -184,15 +184,15 @@ private Module newModule(Class<Module> module) {
                   create)
```

Comment:
Could you refactor the method? It looks that `if (m_testContext.getCurrentXmlTest().isJUnit())` is useless or not a the best place.

Reply:
@juherr - That would need regression (`junit=true` attribute in `<suite>` tag) and it is not related to this bug. So I would not want to include that in this changes and would like to keep the changes as close to fixing the bug as possible.

Reply:
Right, it's outside the goal of fixing bugs but I like to boy scout rule too :p http://www.informit.com/articles/article.aspx?p=1235624&seqNum=6

In fact, I don't understand the logic of the `isJUnit` condition because `result` is overriden later, except in a not obvious case I don't catch.

Reply:
@juherr - Yes I agree with you on the `boy scout rule` and I try to do it as and when time permits. But I generally resist that urge only when the change involves some other functionality wherein I would need to spend time to understand that before making changes, so that I dont break anything. The `isJunit` part is one such area because I have never used TestNG to run any JUnit tests at all.